Cython/Includes/cpython/contextvars.pxd

```diff
@@ -0,0 +1,94 @@
+from cpython.object cimport PyObject
+
+cdef extern from "Python.h":
+
+    ############################################################################
+    # Context Variables Objects
+    ############################################################################
+
+    # PyContext
+    # The C structure used to represent a `contextvars.Context` object.
+
+    # PyContextVar
+    # The C structure used to represent a `contextvars.ContextVar` object.
+
+    # PyContextToken
+    # The C structure used to represent a `contextvars.Token` object.
+
+    # PyTypeObject PyContext_Type
+    # Type object representing the `contextvars.Context` type.
+
+    # PyTypeObject PyContextVar_Type
+    # Type object representing the `contextvars.ContextVar` type.
+
+    # PyTypeObject PyContextToken_Type
+    # Type object representing the `contextvars.Token` type.
+
+    int PyContext_CheckExact(object obj)
+    # Return `true` if `obj` is of type `PyContext_Type`.
+    # `obj` must not be NULL. This function always succeeds.
+
+    int PyContextVar_CheckExact(object obj)
+    # Return `true` if `obj` is of type `PyContextVar_Type`.
+    # `obj` must not be NULL. This function always succeeds.
+
+    int PyContextToken_CheckExact(object obj)
+    # Return `true` if `obj` is of type `PyContextToken_Type`.
+    # `obj` must not be NULL. This function always succeeds.
+
+    object PyContext_New()
+    # Return value: New reference.
+    # Create a new empty context object.
+    # Returns NULL if an error has occurred.
+
+    object PyContext_Copy(object ctx)
+    # Return value: New reference.
+    # Create a shallow copy of the passed `ctx` context object.
+    # Returns NULL if an error has occurred.
+
+    object PyContext_CopyCurrent()
+    # Return value: New reference.
+    # Create a shallow copy of the current thread context.
+    # Returns NULL if an error has occurred.
+
+    int PyContext_Enter(object ctx) except -1
+    # Set `ctx` as the current context for the current thread.
+    # Returns 0 on success, and -1 on error.
+
+    int PyContext_Exit(object ctx) except -1
+    # Deactivate the `ctx` context and restore the previous context
+    # as the current context for the current thread.
+    # Returns 0 on success, and -1 on error.
+
+    int PyContext_ClearFreeList()
+    # Clear the context variable free list.
+    # Return the total number of freed items. This function always succeeds.
+
+    object PyContextVar_New(const char* name, object default_value)
+    # Return value: New reference.
+    # Create a new ContextVar object. The `name` parameter is used
+    # for introspection and debug purposes. The `default_value` parameter
+    # may optionally specify the default value for the context variable.
+    # If an error has occurred, this function returns NULL.
+
+    int PyContextVar_Get(object var, object default_value, PyObject** value) except -1
```

Comment:
It's a bit annoying that the default value in these two functions is allowed to be `NULL` to signal "no default". This cannot be expressed with `object` in Cython. One way would be to use `PyObject *` instead, but that's clumsy to use since it requires a cast on call. However, I guess that many users would want to define context variables also without a default, so this is a use case.

We could provide two functions with different names and the same [cname](http://docs.cython.org/en/latest/src/userguide/external_C_code.html#resolving-naming-conflicts-c-name-specifications), one with `object` that takes a default and the other with `PyObject *` that allows users to easily pass `NULL` for "no default".